modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -792,19 +877,24 @@ private void sendInputMethodEventToFX(InputMethodEvent e) {
      */
     @Override
     protected void paintComponent(Graphics g) {
-        if (scenePeer == null) {
+        EmbeddedSceneInterface lScenePeer = getScenePeer();
+        int lWidth = getCompWidth();
+        int lHeight = getCompHeight();
+        double lScaleFactorX = getScaleFactorX();
+        double lScaleFactorY = getScaleFactorY();
+        if (lScenePeer == null) {
```

Comment:
minor: move check on L885 right after L880, no need to lock 4 times if lScenePeer is null